src/render/tutorials/api-gateway-security/2_0.html.md

```diff
@@ -0,0 +1,954 @@
+
+`BasicAuthHandlerFactory.java`
+```java
+package io.knotx.examples.security.auth;
+
+import io.knotx.server.api.security.AuthHandlerFactory;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.auth.shiro.ShiroAuthOptions;
+import io.vertx.reactivex.core.Vertx;
+import io.vertx.reactivex.ext.auth.shiro.ShiroAuth;
+import io.vertx.reactivex.ext.web.handler.AuthHandler;
+import io.vertx.reactivex.ext.web.handler.BasicAuthHandler;
+
+public class BasicAuthHandlerFactory implements AuthHandlerFactory {
+
+  @Override
+  public String getName() {
+    return "helloBasicAuthFactory";
+  }
+
+  @Override
+  public AuthHandler create(Vertx vertx, JsonObject config) {
+    final ShiroAuth shiroAuth = ShiroAuth.create(vertx, new ShiroAuthOptions().setConfig(config));
+    return BasicAuthHandler.create(shiroAuth);
+  }
+}
+```
+
+It's as simple as creating a [`BasicAuthHandler`](https://vertx.io/docs/apidocs/io/vertx/ext/web/handler/BasicAuthHandler.html) that will use [`ShiroAuth`](https://vertx.io/docs/apidocs/io/vertx/ext/auth/shiro/ShiroAuth.html) instance created with our config.
+
+Let's not forget to register our factory for [Java ServiceLoader](https://docs.oracle.com/javase/7/docs/api/java/util/ServiceLoader.html):
+
+`io.knotx.server.api.security.AuthHandlerFactory`
+```
+io.knotx.examples.security.auth.BasicAuthHandlerFactory
+```
+
+### Build & Run
+
+In this tutorial we'll be using the Docker distribution, but it will work with Stack distribution as well. Please refer to [this tutorial](http://knotx.io/tutorials/getting-started-with-knotx-stack/2_0/) on how to work with the Stack distribution.
+
+First, let's rename the Docker image we're about to create:
+
+`gradle.properties`
+```
+// ...
+docker.image.name=knotx-example/secure-api-gateway
+// ...
+```
+
+Now let's build the image:
+
+```
+$ gradlew clean build-docker
+```
+
+and run it:
+
+```
+$ docker run -p8092:8092 knotx-example/secure-api-gateway
+```
+
+After a while the Docker container should be up and running.
+
+Let's try accessing our endpoints:
+
+```
+$ curl -X GET http://localhost:8092/api/public
+{"message":"Hello World From Knot.x!"}
+```
+
+Our public endpoint is working as expected.
+
+Now let's try the other one:
+
+```
+$ curl -X GET http://localhost:8092/api/secure/basic
+Unauthorized
+```
+
+Oops! We were not authorized to access this endpoint. 
+
+Let's open Postman and construct a request there.
+
+![Basic Auth Postman setup](/img/blog/api-gateway-security/security-postman-basicauth.png)
+
+1. Let's create a new `GET` request for the same URL as before.
+2. Open the `Authorization` tab.
+3. Select `Basic Auth` as the authorization type and type in `john`'s credentials.
+4. Now execute the request.
+
+![Basic Auth Postman response](/img/blog/api-gateway-security/security-postman-basicauth-response.png)
+
+Now the request is successfully authorized and we receive the proper response!
+
+If you don't want to use Postman, you can execute this `curl` command:
```

Comment:
I would explain this: https://docs.apigee.com/api-platform/system-administration/basic-auth#basic-authentication-format

> You can pass your credentials as a Base64-encoded header or as parameters in an HTTP client.
> When you pass your credentials in the header, you must Base64-encode them. The following is an example of an encoded HTTP Basic Authentication header...

Then you would be able to remove postman at all. I think `curl` examples are enough.